src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptorComparer.cs

```diff
@@ -31,7 +31,8 @@ protected TagHelperDescriptorComparer()
         /// <remarks>
         /// Determines equality based on <see cref="TagHelperDescriptor.TypeName"/>,
         /// <see cref="TagHelperDescriptor.AssemblyName"/>, <see cref="TagHelperDescriptor.TagName"/>,
-        /// and <see cref="TagHelperDescriptor.RequiredAttributes"/>.
+        /// <see cref="TagHelperDescriptor.RequiredAttributes"/>, and
+        /// <see cref="TagHelperDescriptor.UsageDescriptor"/>.
```

Comment:
Unlike the `TagHelperAttributeDescriptorComparer` :+1: to this. It can be inferred from both `TypeName` and `AssemblyName`.